lib/ansible/module_utils/facts.py

```diff
@@ -3067,15 +3070,33 @@ def get_file_lines(path):
     return ret
 
 def ansible_facts(module):
+    # Retrieve module parameters
+    gather_subset = [ 'all' ]
+    if 'gather_subset' in module.params:
+        gather_subset = module.params['gather_subset']
+
+    # Retrieve all facts elements
+    if 'all' in gather_subset:
+        gather_subset = [ 'min', 'hardware', 'network', 'virtual' ]
+
+    # Check subsets and forbid unallowed name
+    for subset in gather_subset:
+        if subset not in Facts.ALLOWED_FACT_SUBSET:
+            raise TypeError("Bad subset '%s' given to Ansible. gather_subset options allowed: %s" % (subset, ", ".join(Facts.ALLOWED_FACT_SUBSET)))
+
     facts = {}
     facts.update(Facts().populate())
```

Comment:
Note -- Looks like you have 'min' as a valid subset but in this current version of the code it's not being used to control whether this is run.  Either need to make use of it or remove 'min' as an option so that it's clear that this portion of fact gathering is always run.